### DISCUSSION TITLE
[Mailer] Always Send to the same Address not working as intended?

### DISCUSSION THREAD

**Discussion by tomme87** [symfony/symfony#52592] [Q&A]
With symfony mailer I can, according to the docs; [Always Send to the same Address](https://symfony.com/doc/current/mailer.html#always-send-to-the-same-address). But as far as I can tell, this only works when only using the `to` field or using a smtp-transport.

First have look at [EnvelopeListener](https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Mailer/EventListener/EnvelopeListener.php), this is setting the recipients on the `Envelope`.

Then look at [SendinblueApiTransport](https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Mailer/Bridge/Sendinblue/Transport/SendinblueApiTransport.php) for example. This is using the `Envelope` for the `to` field, but for `cc` and `bcc` this is disregarded. The same applies to other api-transports I have checked.

So if you send a email with `bcc` or `cc` set, this will get sent to the "real" address, and not only to the address configured.

This does not apply to the SMTP-transports, since they will use only `Envelope` for recipients (`Cc` and `To` are only headers).

Is this intended, or am I missing something?
How can I "Always Send to the same Address" also for `cc` and `bcc`?

(Similar issue discussed [here](https://github.com/symfony/symfony/issues/40013))

**Comment by ghost** [comment]
Yes, it looks like a bug. Apart from the purpose of these CC and BCC addresses, they are still addresses that will receive the email. So if we use `Always send to the same address` feature then it should not send the email to any other addresses than the ones configured.
